### PR TITLE
feat: add rpc methods

### LINKED ISSUE #3372: Add API for fresh taproot address
Allow wallet to request new addresses.

**use case**
App want to send an inscription to user. It needs address to do so. It calls method that opens up pop up for the user to accept/reject.

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/wallet/actions/runs/4490852534).

RPC structure to support adding new methods more easily, without JWT.

Following pattern where each new RPC call is added as a new file under `rpc/methods` folder. Exporting an interface of a single key of the method name, containing request/response objects. This is used to later infer the types.

```ts
const accounts = await window.btc?.request('getAddresses');
```

### CODE REVIEW

File: src/app/components/favicon.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+interface FaviconProps {
+  origin: string;
+}
+
+export function Favicon({ origin }: FaviconProps) {
+  const [hasError, setHasError] = useState(false);
+  if (hasError) return null;
+  return (
+    <img
+      src={`http://www.google.com/s2/favicons?domain=${origin}`}
```
Comment (edu-stx) [inline src/app/components/favicon.tsx:12]: Very cool!

File: src/app/pages/rpc-get-addresses/components/get-addresses.layout.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      </Flex>
+      <Flex
+        backgroundColor="#F5F5F7"
+        px="loose"
+        py="base"
+        lineHeight="20px"
+        textAlign="center"
+        alignSelf="bottom"
+      >
+        <Text fontSize="14px" color="#74777D">
```
Comment (edu-stx) [inline src/app/pages/rpc-get-addresses/components/get-addresses.layout.tsx]:
```suggestion
        <Text fontSize="14px" color={figmaTheme.textSubdued}>
```

File: src/app/pages/rpc-get-addresses/components/get-addresses.layout.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        </Flag>
+
+        <Button mt="extra-loose" onClick={() => onUserApproveGetAddresses()} width="100%">
+          <Flag align="middle" img={<img src={HiroLogoInBox} width="16px" />}>
+            Connect to Hiro Wallet
+          </Flag>
+        </Button>
+      </Flex>
+      <Flex
+        backgroundColor="#F5F5F7"
```
Comment (edu-stx) [inline src/app/pages/rpc-get-addresses/components/get-addresses.layout.tsx]: figmaTheme.backgroundsubdued

(I forgot to camel case this one)

**Review by fbwoolf — APPROVED** [review]
Looks great, @kyranjamie. We are still leaving the one address provided in auth response?

File: src/app/store/accounts/blockchain/bitcoin/bitcoin-account.models.ts [hunk 4]
```diff
@@ -25,9 +28,14 @@ export const tempHardwareAccountForTesting: HardwareBitcoinAccount = {
 };
 
 export function formatBitcoinAccount(keychain: HDKey) {
+  if (keychain.depth !== DerivationPathDepth.Account)
+    throw new Error('Can only format from account keychain');
   return (index: number): SoftwareBitcoinAccount => ({
     type: 'software',
     index,
+    // The rationate for wrapping the keychain is so we pass around the xpub
```
Comment (edu-stx) [inline src/app/store/accounts/blockchain/bitcoin/bitcoin-account.models.ts]:
```suggestion
    // The rationale for wrapping the keychain is so we pass around the xpub
```

File: src/app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks.ts [hunk 5]
```diff
@@ -31,6 +31,22 @@ function useNativeSegWitCurrentNetworkAccountKeychain() {
   );
 }
 
+function useCurrentBitcoinNativeSegwitAccountPublicKeychain() {
+  const { xpub } = useCurrentBitcoinNativeSegwitAccountInfo();
+  if (!xpub) return; // TODO: Revisit this return early
+  const keychain = HDKey.fromExtendedKey(xpub);
+  if (!keychain?.publicKey) throw new Error('No public key for given keychain');
```
Comment (edu-stx) [inline src/app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks.ts]: Getting an "Unnecessary optional chaining" warning here, is it warranted?

**Comment by kyranjamie** [comment]
I have removed a `pb` prop from our `Container` component FYI @edu-stx @fbwoolf @alter-eggo. We can't assume this will always need padding at bottom, but removes padding where it's needed, let's just add back explicitly to each feature that needs it

**Review by edu-stx — APPROVED** [review]

**Pull request merged**